samples/matter/window_covering/src/zcl_callbacks.cpp

```diff
@@ -67,3 +67,55 @@ void MatterWindowCoveringClusterServerAttributeChangedCallback(const app::Concre
 		};
 	}
 }
+
+void emberAfWindowCoveringClusterInitCallback(chip::EndpointId endpoint)
+{
+	const auto logOnFailure = [](EmberAfStatus status, const char *attributeName) {
+		if (status != EMBER_ZCL_STATUS_SUCCESS) {
+			ChipLogError(Zcl, "Failed to set WindowCovering %s: %x", attributeName, status);
+		}
+	};
+
+	app::DataModel::Nullable<chip::Percent100ths> currentPercent100ths;
+	app::DataModel::Nullable<chip::Percent100ths> targetPercent100ths;
+	app::DataModel::Nullable<chip::Percent> currentPercentage;
+	EmberAfStatus status;
+
+	status = Attributes::CurrentPositionLiftPercentage::Get(endpoint, currentPercentage);
+	if (currentPercentage.IsNull()) {
+		logOnFailure(Attributes::CurrentPositionLiftPercentage::Set(endpoint, 0),
+			     "current position lift percentage");
+	}
+
+	status = Attributes::CurrentPositionTiltPercentage::Get(endpoint, currentPercentage);
+	if (currentPercentage.IsNull()) {
+		logOnFailure(Attributes::CurrentPositionTiltPercentage::Set(endpoint, 0),
+			     "current position tilt percentage");
+	}
+
+	status = Attributes::CurrentPositionLiftPercent100ths::Get(endpoint, currentPercent100ths);
+	if (currentPercent100ths.IsNull()) {
+		currentPercent100ths.SetNonNull(0);
+		logOnFailure(Attributes::CurrentPositionLiftPercent100ths::Set(endpoint, 0),
+			     "current position lift percent 100ths");
+	}
+
+	status = Attributes::TargetPositionLiftPercent100ths::Get(endpoint, targetPercent100ths);
+	if (targetPercent100ths.IsNull()) {
```

Comment:
It's not bad naming, it's intentional :smile: My concern was the situation that for some reason after boot we have current position set to non-null value (it can be 0 or anything else) and we have target position set to null. In such situation we have to initialize target position with some non-null value, so I think the best option is to use current position, so in the end window covering will not change the position. Answering on your comment above, that's why I set `currentPercent100ths` to 0, so this variable can be used to pass 0 or non-zero value read from ZCL to target position.

Reply:
Ok, I see. But the fact that we need to set the target position to something else than null is actually a part of this fix, right? I mean, from the device/spec perspective I think this is reasonable to have the target position set to null after boot but we need to change that to be compatible with ecosystems.

Reply:
Discussed F2F